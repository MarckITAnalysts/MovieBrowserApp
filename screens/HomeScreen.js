import {
  View,
  Text,
  TouchableOpacity,
  ScrollView,
  Platform,
} from "react-native";
import React, { useEffect, useState } from "react";
import { SafeAreaView } from "react-native-safe-area-context";
import {
  MagnifyingGlassIcon,
} from "react-native-heroicons/outline";
import TrendingMovies from "../components/trendingMovies";
import MovieList from "../components/movieList";
import { StatusBar } from "expo-status-bar";
import {
  fetchTopRatedMovies,
  fetchNowPlayingMovies,
  fetchTrendingMovies,
  fetchUpcomingMovies,
} from "../api/moviedb";
import { useNavigation } from "@react-navigation/native";
import Loading from "../components/loading";
import { styles } from "../theme";
//Added before because api is not working so gettind data from json
import trendingMovies from "../response/popular.json";
import upcomingMovies from "../response/upcoming.json";
import topRatedMovies from "../response/top_rated.json";
import nowPlayingMovies from "../response/now_playing.json";

const ios = Platform.OS === "ios";

export default function HomeScreen() {
  const [trending, setTrending] = useState([]);
  const [upcoming, setUpcoming] = useState([]);
  const [topRated, setTopRated] = useState([]);
  const [nowPlaying, setNowPlaying] = useState([]);
  const [loading, setLoading] = useState(true);
  const navigation = useNavigation();

  useEffect(() => {
    getTrendingMovies();
    getUpcomingMovies();
    getTopRatedMovies();
    getNowPlayingMovies();
  }, []);

  const getTrendingMovies = async () => {
    const data = await fetchTrendingMovies();
    console.log("got trending::", data);
    if (data && data.results) setTrending(data.results);
    setLoading(false);
  };
  const getUpcomingMovies = async () => {
    const data = await fetchUpcomingMovies();
    console.log("got upcoming::", data);
    if (data && data.results) setUpcoming(data.results);
  };
  const getTopRatedMovies = async () => {
    const data = await fetchTopRatedMovies();
    console.log("got top rated::", data,);
    if (data && data.results) setTopRated(data.results);
  };

  const getNowPlayingMovies = async () => {
    const data = await fetchNowPlayingMovies();
    console.log("got now playing::", data);
    if (data && data.results) setNowPlaying(data.results);
    setLoading(false);
  };

  return (
    <View className="flex-1 bg-neutral-800">
      {/* search bar */}
      <SafeAreaView className={ios ? "-mb-2" : "mb-3"}>
        <StatusBar style="light" />
        <View className="flex-row justify-between items-center mx-4">
          <Text className="text-white text-3xl font-bold">
            <Text style={styles.text}>M</Text>ovies
          </Text>
          <TouchableOpacity onPress={() => navigation.navigate("Search")}>
            <MagnifyingGlassIcon size="30" strokeWidth={2} color="white" />
          </TouchableOpacity>
        </View>
      </SafeAreaView>
      {loading ? (
        <Loading />
      ) : (
        <ScrollView
          showsVerticalScrollIndicator={false}
          contentContainerStyle={{ paddingBottom: 10 }}
        >

          {/* Trending Movies Carousel */}
          { trending?.length>0 && <TrendingMovies data={trending} /> }

          {/* upcoming movies row */}
          {nowPlaying?.length > 0 && ( <MovieList title="Now Playing" data={nowPlaying} />)}

          {/* upcoming movies row */}
          { upcoming?.length>0 && <MovieList title="Upcoming" data={upcoming} /> }
            
          {/* top rated movies row */}
          { topRated?.length>0 && <MovieList title="Top Rated" data={topRated} /> }
        </ScrollView>
      )}
    </View>
  );
}
