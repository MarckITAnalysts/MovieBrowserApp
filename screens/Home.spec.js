import React from 'react';
import renderer, {act} from 'react-test-renderer';
import HomeScreen from './HomeScreen';

jest.useFakeTimers()

jest.mock('react-native-snap-carousel', () => 'Carousel');
jest.mock('expo-status-bar', () => ({
  ...jest.requireActual('expo-status-bar'),
  StatusBar: () => {},
}));
jest.mock('react-native-progress', () => ({
  ...jest.requireActual('react-native-progress'),
  Progress: () => {},
}));

const mockedUsedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => mockedUsedNavigate,
}));

jest.mock('React', () => ({
  ...jest.requireActual('React'),
  useEffect: jest.fn(),
}));


describe('Home screen test cases', () => {
  beforeEach(() => {
    React.useState = jest
      .fn()
      .mockReturnValueOnce([[], {}])
      .mockReturnValueOnce([[], {}])
      .mockReturnValueOnce([[], {}])
      .mockReturnValueOnce([[], {}]);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
