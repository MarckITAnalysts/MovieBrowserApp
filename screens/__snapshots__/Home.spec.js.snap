// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screet test cases renders correctly 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#262626",
      },
    ]
  }
>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      [
        {
          "marginBottom": -8,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "justifyContent": "space-between",
          },
          {
            "alignItems": "center",
          },
          {
            "marginLeft": 16,
            "marginRight": 16,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#fff",
            },
            {
              "fontSize": 30,
              "lineHeight": 36,
            },
            {
              "fontWeight": "700",
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
            }
          }
        >
          M
        </Text>
        ovies
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          aria-hidden="true"
          bbHeight="30"
          bbWidth="30"
          color="white"
          fill="none"
          focusable={false}
          height="30"
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="currentColor"
          strokeWidth={2}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 30,
                "width": 30,
              },
            ]
          }
          tintColor="white"
          vbHeight={24}
          vbWidth={24}
          width="30"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeWidth="2"
          >
            <RNSVGPath
              d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              strokeLinecap={1}
              strokeLinejoin={1}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </RNCSafeAreaView>
  <View
    style={
      [
        {
          "position": "absolute",
        },
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "center",
        },
        {
          "alignItems": "center",
        },
        {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "overflow": "hidden",
          "transform": [
            {
              "rotate": "0deg",
            },
          ],
        }
      }
      useNativeDriver={false}
    >
      <RNSVGSvgView
        bbHeight="160"
        bbWidth="160"
        focusable={false}
        height={160}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 160,
              "width": 160,
            },
          ]
        }
        width={160}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGPath
            d="M73.81032816789664 18.309741752762406
          A62 62 0 0 1 80 18"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
              ]
            }
            stroke={
              {
                "payload": 4293571336,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeWidth="12"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
